controller/src/conf/application.conf

```diff
@@ -81,6 +81,13 @@ config {
     transaction {
       minLeaseValue = ${MIN_LEASE_VALUE}
       maxLeaseValue = ${MAX_LEASE_VALUE}
+      completed {
+        ttlInHours = ${COMPLETED_TXN_TTL_IN_HOURS}
+        #Before 0.3.1 all completed transaction records are written to old scheme path and never gc'd.
+        #If this flag is enabled, controller will write completed transaction record to both old and new scheme paths.
+        #Disable this flag after all controller instances have been upgraded to version 0.3.1 and above.
+        disableBackwardCompatiblity = ${DISABLE_BACKWARD_COMPATIBLITY}
```

Comment:
I'm wondering about the lifecycle of this option. Say I'm upgrading from 0.3.1 to 0.4.0. As I bounce controller instances, I leave backward compatibility enabled. Once I finish bouncing all instances, should I bounce them again one by one disabling the backward compatibility?

Reply:
yes, typically after a successful upgrade and once at least one batch interval has elapsed, we can safely disable this flag in rolling update fashion.
One batch interval because then we are sure that all transactions that were written only to old path during upgrade have expired and there would be no need to look at old path anymore.